File: output/page_file_generator/old_files/Equfix_Place_On_Alert_palywright.py
from playwright.sync_api import Page


class equifax_actions:

    def __init__(self, page: Page):
        self.page = page
        self.continue_button = page.locator("xpath=//button[@id='continue-button']")
        self.zip_code = page.locator("xpath=//input[@id='zip']")
        self.dropdown_label = page.locator("xpath=//button[@id='efx-dropdown-label-753393']")
        self.city = page.locator("xpath=//input[@id='city']")
        self.address_line_2 = page.locator("xpath=//input[@id='addressLine2Id']")
        self.address_line_1 = page.locator("xpath=//input[@id='address']")
        self.phone_number = page.locator("xpath=//input[@id='phoneNumber']")
        self.ssn = page.locator("xpath=//input[@id='ssn']")
        self.date_of_birth = page.locator("xpath=//input[@id='dob']")
        self.last_name = page.locator("xpath=//input[@id='lastName']")
        self.first_name = page.locator("xpath=//input[@id='firstNameId']")

    def wait_for_element(self, locator):
        locator.wait_for()

    def click(self, locator):
        self.wait_for_element(locator)
        locator.click()

    def enter_text(self, locator, text):
        self.wait_for_element(locator)
        locator.fill(text)

    def switch_to_window_by_index(self, index):
        all_pages = self.page.context.pages
        if index < 0 or index >= len(all_pages):
            raise IndexError("Invalid window index.")
        self.page = all_pages[index]
        self.page.bring_to_front()

    def switch_to_window_by_handle(self, handle):
        all_pages = self.page.context.pages
        for page in all_pages:
            if page == handle:
                self.page = page
                self.page.bring_to_front()
                return
        raise RuntimeError(f"No window found for handle: {handle}")

    def switch_to_new_window(self, url=None):
        with self.page.context.expect_page() as new_page:
            pass
        new_page = new_page.value
        new_page.bring_to_front()
        if url and new_page.url != url:
            raise RuntimeError(f"New window URL '{new_page.url}' does not match expected URL '{url}'")
        self.page = new_page

    def switch_to_window_matching_url(self, url):
        all_pages = self.page.context.pages
        for page in all_pages:
            if page.url == url:
                self.page = page
                self.page.bring_to_front()
                return
        raise RuntimeError(f"No window found with URL: {url}")

    def click_continue_button(self):
        self.click(self.continue_button)

    def enter_zip(self, text):
        self.enter_text(self.zip_code, text)

    def click_dropdown_label(self):
        self.click(self.dropdown_label)

    def enter_city(self, text):
        self.enter_text(self.city, text)

    def enter_address_line_2(self, text):
        self.enter_text(self.address_line_2, text)

    def enter_address_line_1(self, text):
        self.enter_text(self.address_line_1, text)

    def enter_phone_number(self, text):
        self.enter_text(self.phone_number, text)

    def click_ssn(self):
        self.click(self.ssn)

    def enter_ssn(self, text):
        self.enter_text(self.ssn, text)

    def enter_last_name(self, text):
        self.enter_text(self.last_name, text)

    def enter_first_name(self, text):
        self.enter_text(self.first_name, text)

    def enter_date_of_birth(self, text):
        self.enter_text(self.date_of_birth, text)

File: output/page_file_generator/old_files/Equfix_Place_On_Alert_palywright.java
import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.BrowserContext;
import com.microsoft.playwright.Page.WaitForSelectorOptions;

import java.util.ArrayList;
import java.util.List;

public class EquifixFlowActions {

    private Page page;

    private Locator continueButton;
    private Locator zip;
    private Locator efxDropdownLabel;
    private Locator city;
    private Locator addressLine2;
    private Locator address;
    private Locator phoneNumber;
    private Locator ssn;
    private Locator dob;
    private Locator lastName;
    private Locator firstNameId;

    public EquifixFlowActions(Page page) {
        this.page = page;
        this.continueButton = page.locator("//button[@id='continue-button']");
        this.zip = page.locator("//input[@id='zip']");
        this.efxDropdownLabel = page.locator("//button[@id='efx-dropdown-label-753393']");
        this.city = page.locator("//input[@id='city']");
        this.addressLine2 = page.locator("//input[@id='addressLine2Id']");
        this.address = page.locator("//input[@id='address']");
        this.phoneNumber = page.locator("//input[@id='phoneNumber']");
        this.ssn = page.locator("//input[@id='ssn']");
        this.dob = page.locator("//input[@id='dob']");
        this.lastName = page.locator("//input[@id='lastName']");
        this.firstNameId = page.locator("//input[@id='firstNameId']");
    }

    private void waitForElement(Locator locator) {
        locator.waitFor(new WaitForSelectorOptions().setState(WaitForSelectorOptions.State.ATTACHED));
    }

    private void click(Locator locator) {
        locator.click();
    }

    private void enterText(Locator locator, String text) {
        locator.fill(text);
    }

    private void switchToWindowByIndex(int index) {
        List<Page> pages = new ArrayList<>(page.context().pages());
        if (index >= pages.size()) {
            throw new RuntimeException("Invalid window index: " + index);
        }
        page = pages.get(index);
    }

    private void switchToWindowByHandle(String handle) {
        List<Page> pages = new ArrayList<>(page.context().pages());
        for (Page p : pages) {
            if (p.mainFrame().url().equals(handle)) {
                page = p;
                return;
            }
        }
        throw new RuntimeException("No window with handle: " + handle);
    }

    private void switchToWindowMatchingUrl(String url) {
        List<Page> pages = new ArrayList<>(page.context().pages());
        for (Page p : pages) {
            if (p.url().equals(url)) {
                page = p;
                return;
            }
        }
        throw new RuntimeException("No window with URL: " + url);
    }

    private void switchToNewWindow(String url) {
        List<Page> pages = new ArrayList<>(page.context().pages());
        page.context().waitForEvent("page");
        Page newPage = pages.get(pages.size() - 1);
        if (!newPage.url().equals(url)) {
            throw new RuntimeException("New window with URL not found: " + url);
        }
        page = newPage;
    }

    public void enterFirstName(String firstName) {
        waitForElement(firstNameId);
        enterText(firstNameId, firstName);
    }
    
    public void clickSSNField() {
        waitForElement(ssn);
        click(ssn);
    }
    
    public void enterSSN(String ssnValue) {
        waitForElement(ssn);
        enterText(ssn, ssnValue);
    }

    public void clickLastNameField() {
        waitForElement(lastName);
        click(lastName);
    }

    public void enterLastName(String lastNameValue) {
        waitForElement(lastName);
        enterText(lastName, lastNameValue);
    }

    public void clickPhoneNumberField() {
        waitForElement(phoneNumber);
        click(phoneNumber);
    }

    public void enterPhoneNumber(String phoneValue) {
        waitForElement(phoneNumber);
        enterText(phoneNumber, phoneValue);
    }

    public void clickDOBField() {
        waitForElement(dob);
        click(dob);
    }

    public void enterDOB(String dobValue) {
        waitForElement(dob);
        enterText(dob, dobValue);
    }

    public void clickAddressField() {
        waitForElement(address);
        click(address);
    }

    public void enterAddress(String addressValue) {
        waitForElement(address);
        enterText(address, addressValue);
    }

    public void clickCityField() {
        waitForElement(city);
        click(city);
    }

    public void enterCity(String cityValue) {
        waitForElement(city);
        enterText(city, cityValue);
    }

    public void clickAddressLine2Field() {
        waitForElement(addressLine2);
        click(addressLine2);
    }

    public void enterAddressLine2(String addressLine2Value) {
        waitForElement(addressLine2);
        enterText(addressLine2, addressLine2Value);
    }

    public void clickZipCodeField() {
        waitForElement(zip);
        click(zip);
    }

    public void enterZipCode(String zipCodeValue) {
        waitForElement(zip);
        enterText(zip, zipCodeValue);
    }

    public void clickContinueButton() {
        waitForElement(continueButton);
        click(continueButton);
    }

    public void selectState(String state) {
        waitForElement(efxDropdownLabel);
        click(efxDropdownLabel);
        Locator stateOption = page.locator("//button[@id='efx-dropdown-label-" + state + "']");
        waitForElement(stateOption);
        click(stateOption);
    }
}